src/db/pool.cr

```diff
@@ -47,6 +57,49 @@ module DB
       @initial_pool_size.times { build_resource }
     end
 
+    def reap_connection!(connection : T, delay : Float64)
+      sleep delay
+
+      sync do
+        connection_idle = @idle.includes? connection
+        return unless connection_idle # let the next reaping attempt get it
```

Comment:
alternatively, you could reschedule for a later interval (500ms?) and have a form of exponential backoff on retry on future spawns